types/systemd/servicename.pp

```diff
@@ -0,0 +1,2 @@
+# Valid systemd service names
+type Simplib::Systemd::ServiceName = Pattern['^([A-za-z0-9.:_\\-]){1,256}$']
```

Comment:
Should this be capital Z here in the regex? i.e.

type Simplib::Systemd::ServiceName = Pattern['^([A-Za-z0-9.:_\\-]){1,256}$']